lib/MainTemplate.js

```diff
@@ -485,10 +487,19 @@ module.exports = class MainTemplate extends Tapable {
 	 * @returns {string} hook call
 	 */
 	getPublicPath(options) {
-		return this.hooks.assetPath.call(
-			this.outputOptions.publicPath || "",
-			options
-		);
+		let publicPath = "";
+		if (this.devServerOptions.host) {
+			publicPath = `${this.outputOptions.https ? "https" : "http"}://${
+				this.devServerOptions.host
+			}`;
+		}
+		if (this.devServerOptions.port) {
+			publicPath += `:${this.devServerOptions.port}`;
+		}
+		if (this.devServerOptions.publicPath) {
+			publicPath += this.outputOptions.publicPath;
+		}
```

Comment:
This could be resolved at a higher level, for example in `webpack.js`. So we don't inject the devServer config here, and the publicPath contains all the information.
As I am not deeply familiar with the code, I would love some pointers.
Please let me know what's your take on this.